service/history/task/task.go

```diff
@@ -67,18 +67,19 @@ type (
 		sync.Mutex
 		Info
 
-		shard         shard.Context
-		state         ctask.State
-		priority      int
-		attempt       int
-		timeSource    clock.TimeSource
-		submitTime    time.Time
-		logger        log.Logger
-		scopeIdx      int
-		emitDomainTag bool
-		scope         metrics.Scope // initialized when processing task to make the initialization parallel
-		taskExecutor  Executor
-		maxRetryCount dynamicconfig.IntPropertyFn
+		shard             shard.Context
+		state             ctask.State
+		priority          int
+		attempt           int
+		timeSource        clock.TimeSource
+		submitTime        time.Time
+		logger            log.Logger
+		scopeIdx          int
+		emitDomainTag     bool
```

Comment:
Let's remove this flag and it's corresponding dynamic config given that it won't affect performance.